### PR TITLE
BUGFIX: Fix PluginImplementation unit test

### PR DESCRIPTION
**Pull request by kdambekalns**
This adjusts the test to the PSR-7 changes.

### CODE REVIEW

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 1]
```diff
@@ -41,25 +44,44 @@ class PluginImplementationTest extends UnitTestCase
...
+     */
+    protected $mockDispatcher;
 
     public function setUp(): void
     {
         $this->pluginImplementation = $this->getAccessibleMock(PluginImplementation::class, ['buildPluginRequest'], [], '', false);
 
         $this->mockHttpUri = $this->getMockBuilder(Uri::class)->disableOriginalConstructor()->getMock();
-        $this->mockHttpUri->expects(self::any())->method('getHost')->will(self::returnValue('localhost'));
+        $this->mockHttpUri->expects(self::any())->method('getHost')->willReturn('localhost');
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockHttpUri->method('getHost')->willReturn('localhost');
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 2]
```diff
@@ -41,25 +44,44 @@ class PluginImplementationTest extends UnitTestCase
...
 
-        $this->mockHttpRequest = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->getMock();
-        $this->mockHttpRequest->expects(self::any())->method('getUri')->will(self::returnValue($this->mockHttpUri));
+        $this->mockHttpRequest = $this->getMockBuilder(HttpRequestInterface::class)->disableOriginalConstructor()->getMock();
+        $this->mockHttpRequest->expects(self::any())->method('getUri')->willReturn($this->mockHttpUri);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockHttpRequest->method('getUri')->willReturn($this->mockHttpUri);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 3]
```diff
@@ -41,25 +44,44 @@ class PluginImplementationTest extends UnitTestCase
...
 
         $this->mockActionRequest = $this->getMockBuilder(ActionRequest::class)->disableOriginalConstructor()->getMock();
-        $this->mockActionRequest->expects(self::any())->method('getHttpRequest')->will(self::returnValue($this->mockHttpRequest));
+        $this->mockActionRequest->expects(self::any())->method('getHttpRequest')->willReturn($this->mockHttpRequest);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockActionRequest->method('getHttpRequest')->willReturn($this->mockHttpRequest);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 4]
```diff
@@ -41,25 +44,44 @@ class PluginImplementationTest extends UnitTestCase
...
 
         $this->mockControllerContext = $this->getMockBuilder(ControllerContext::class)->disableOriginalConstructor()->getMock();
-        $this->mockControllerContext->expects(self::any())->method('getRequest')->will(self::returnValue($this->mockActionRequest));
+        $this->mockControllerContext->expects(self::any())->method('getRequest')->willReturn($this->mockActionRequest);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockControllerContext->method('getRequest')->willReturn($this->mockActionRequest);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 5]
```diff
@@ -41,25 +44,44 @@ class PluginImplementationTest extends UnitTestCase
...
 
         $this->mockRuntime = $this->getMockBuilder(Runtime::class)->disableOriginalConstructor()->getMock();
-        $this->mockRuntime->expects(self::any())->method('getControllerContext')->will(self::returnValue($this->mockControllerContext));
+        $this->mockRuntime->expects(self::any())->method('getControllerContext')->willReturn($this->mockControllerContext);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockRuntime->method('getControllerContext')->willReturn($this->mockControllerContext);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 6]
```diff
@@ -90,43 +112,42 @@ public function responseHeadersDataprovider()
...
      *
-     * @dataProvider responseHeadersDataprovider
+     * @dataProvider responseHeadersDataProvider
      * @test
      */
-    public function evaluateSetHeaderIntoParent($message, $input, $expected)
+    public function evaluateSetHeaderIntoParent(string $message, array $input, array $expected): void
     {
-        $this->pluginImplementation->expects(self::any())->method('buildPluginRequest')->will(self::returnValue($this->mockActionRequest));
+        $this->pluginImplementation->expects(self::any())->method('buildPluginRequest')->willReturn($this->mockActionRequest);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->pluginImplementation->method('buildPluginRequest')->willReturn($this->mockActionRequest);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 7]
```diff
@@ -90,43 +112,42 @@ public function responseHeadersDataprovider()
...
 
-        $parentResponse = new \GuzzleHttp\Psr7\Response();
+        $parentResponse = new ActionResponse();
         $this->_setHeadersIntoResponse($parentResponse, $input['parent']);
-        $this->mockControllerContext->expects(self::any())->method('getResponse')->will(self::returnValue($parentResponse));
+        $this->mockControllerContext->expects(self::any())->method('getResponse')->willReturn($parentResponse);
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockControllerContext->method('getResponse')->willReturn($parentResponse);
```

File: Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php [hunk 8]
```diff
@@ -90,43 +112,42 @@ public function responseHeadersDataprovider()
...
 
-        $this->mockDispatcher->expects(self::any())->method('dispatch')->will(self::returnCallback(function ($request, $response) use ($input) {
-            $response = $this->_setHeadersIntoResponse($response, $input['plugin']);
-        }));
+        $this->mockDispatcher->expects(self::any())->method('dispatch')->willReturnCallback(function (ActionRequest $request, ActionResponse $response) use ($input) {
```
Comment (bwaidelich) [inline Neos.Neos/Tests/Unit/Fusion/PluginImplementationTest.php]:
```suggestion
        $this->mockDispatcher->method('dispatch')->willReturnCallback(function (ActionRequest $request, ActionResponse $response) use ($input) {
```

**Comment by kdambekalns** [comment]
@bwaidelich I saw PhpStorm complain, but didn't know you can set up behaviour without `expect()`ing anything…

**Pull request merged**